cpp/config/Make.rules

```diff
@@ -99,8 +99,14 @@ endif
 #
 cpp11_cppflags          = -DICE_CPP11_MAPPING
 ifneq ($(shell $(CC) -E config/cplusplus_check.cpp 1> /dev/null 2>&1; echo $$?),0)
+ifeq ($(os),Darwin)
+# With Xcode 14.3 C++17 deprecations apply when building with -std=c++11, we pass -std=c++17
```

Comment:
What is the default -std with the clang versions we support on macos? 
If it is c++17 or greater, I would simply remove the -std flag.

The main reason we have this `=std=c++11` is old compiler's default std was < c++11. Must be rare now.

Reply:
The `__cplusplus` macro defaults to `199711L`
```
echo | clang++ -dM -E -x c++ - | grep cplus
#define __cplusplus 199711L
```

Reply:
Ok.